### DISCUSSION TITLE
react gatsby typescript not enforcing

### DISCUSSION THREAD

**Discussion by kylewinkler** [gatsbyjs/gatsby#38852] [Help]
Hey all.
I am working on a fresh react-gatsby typescript project.
ts version: 5.3.3

My project is compiling without error with existing implicit any declarations.
My IDE actually errors on the problem code, but the project compiles and runs just fine. Any thoughts??

I have the following tsconfig.json:

`{
  "compilerOptions": {
    "target": "es5",
    "lib": ["dom", "dom.iterable", "esnext"],
    "noImplicitAny": true,
    "allowJs": true,
    "skipLibCheck": true,
    "esModuleInterop": true,
    "allowSyntheticDefaultImports": true,
    "strict": true,
    "forceConsistentCasingInFileNames": true,
    "noFallthroughCasesInSwitch": true,
    "module": "esnext",
    "moduleResolution": "node",
    "resolveJsonModule": true,
    "isolatedModules": true,
    "noEmit": true,
    "jsx": "react-jsx",
    "baseUrl": "src",
  },
  "include": ["src", "gatsby-config.ts"],
}
`

and the following exists in my gatsby-config.ts:

`
  plugins: [
    'gatsby-plugin-typescript',
  ...
`

**Comment by AndrewKMoses** [comment]
What are you referring to when you say "compile"?

Typescript type checking / build and Gatsbyjs build are two different things. Gatsbyjs build will not detect typescript issues.
